# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpConnection.h
# Commit: e9635ec4198f
# Full Hash: e9635ec4198fb5293c6509fce2fa42775de0d71e
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-22 15:53:06
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Refactor EnsureNPNComplete r=necko-reviewers,kershaw
#   
#   Extract Check0RttEnabled
#   The old code checks 0RTT state then does a DriveHandshake then checks 0RTT again. This is done in this way because before DriveHandshake is called for the first time 0RTT states are not set.  DriveHandshake is sometimes called as a side effect by IsAlive() check. The new code makes this less complex and just calls DriveHandshaek before checking 0RTT.
#   Extract code for setting 0RTT telemetry values.
# ==============================================================================

diff -r a069ce12654e -r e9635ec4198f netwerk/protocol/http/nsHttpConnection.h
--- a/netwerk/protocol/http/nsHttpConnection.h	Wed Sep 22 09:13:24 2021 +0000
+++ b/netwerk/protocol/http/nsHttpConnection.h	Wed Sep 22 09:19:29 2021 +0000
@@ -243,6 +243,8 @@
   [[nodiscard]] nsresult DisableTCPKeepalives();
 
   bool CheckCanWrite0RTTData();
+  void Check0RttEnabled(nsISSLSocketControl* ssl);
+  void EarlyDataTelemetry(int16_t tlsVersion, bool earlyDataAccepted);
 
  private:
   // mTransaction only points to the HTTP Transaction callbacks if the
